client/pd_service_discovery.go

```diff
@@ -198,56 +201,58 @@ func (c *pdBaseClient) memberLoop() {
 }
 
 // Close releases all resources
-func (c *pdBaseClient) Close() {
-	c.clientConns.Range(func(key, cc interface{}) bool {
-		if err := cc.(*grpc.ClientConn).Close(); err != nil {
-			log.Error("[pd] failed to close gRPC clientConn", errs.ZapError(errs.ErrCloseGRPCConn, err))
-		}
-		c.clientConns.Delete(key)
-		return true
+func (c *pdServiceDiscovery) Close() {
+	c.closeOnce.Do(func() {
```

Comment:
I find here we use `once`, does that mean we might do `close` more than one time?

Reply:
As the same answer to init() above